core/flyout_base.js

```diff
@@ -219,7 +219,7 @@ Blockly.Flyout.prototype.DRAG_RADIUS = 10;
  * @type {number}
  * @const
  */
-Blockly.Flyout.prototype.MARGIN = 12;
+Blockly.Flyout.prototype.MARGIN = 4;
```

Comment:
You probably want to Change GAP_Y (line 236).  Changing MARGIN has more impact on layout.